### ISSUE TITLE
default backend 404 error 

### ISSUE THREAD

**Issue by rancher-max** [rancher/rio#602]
**Describe the bug**
Running GH images on v0.6.0-alpha.1 fails with `ImagePullBackOff/Back-off pulling image "localhost:5442/default-gh:19bc3`

**To Reproduce**
1. Install rio into a k3s v0.10.0 cluster
2. Copy and paste the command that is output from install: `rio run https://github.com/rancher/rio-demo`
After the build succeeds, the service has a scale of 0/1 and shows the error above in the details

**Expected behavior**
It should be up and able to be queried.

**Kubernetes version & type (GKE, on-prem)**: `kubectl version`
```
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.0", GitCommit:"2bd9643cee5b3b3a5ecbd3af49d09018f0773c77", GitTreeState:"clean", BuildDate:"2019-09-19T13:57:45Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.2-k3s.1", GitCommit:"b8b17ba55f20e590df507fce333dfee13ab438c6", GitTreeState:"clean", BuildDate:"2019-10-16T05:17Z", GoVersion:"go1.13.3", Compiler:"gc", Platform:"linux/amd64"}
```
Type:
**Rio version**: `rio info`
```
Rio Version: v0.6.0-alpha.1 (7678d67d)
Rio CLI Version: v0.6.0-alpha.1 (7678d67d)
Cluster Domain: xxxxxx.on-rio.io
Cluster Domain IPs: x.x.x.x
System Namespace: rio-system
System Ready State: false
Wildcard certificates: true
```
### Current status
* The local build issue is being fixed by k3s team.

**Comment by StrongMonkey** [comment]
It turns out that hostNetwork is not working in k3s v0.10.0. We deployed our registry into cluster and registry proxy deploy as HostNetwork to true and listen on localhost:5442.

**Comment by StrongMonkey** [comment]
We should verify if k8s 1.16 on other distros works.

**Comment by rancher-max** [comment]
Tried again with a non-k3s setup. This time, it was able to pull the image and gave an endpoint, but hitting the endpoint gave one of two errors: `default backend - 404` OR
```
curl: (60) SSL certificate problem: unable to get local issuer certificate
More details here: https://curl.haxx.se/docs/sslcerts.html

curl performs SSL certificate verification by default, using a "bundle"
 of Certificate Authority (CA) public keys (CA certs). If the default
 bundle file isn't adequate, you can specify an alternate file
 using the --cacert option.
If this HTTPS server uses a certificate signed by a CA represented in
 the bundle, the certificate verification probably failed due to a
 problem with the certificate (it might be expired, or the name might
 not match the domain name in the URL).
If you'd like to turn off curl's verification of the certificate, use
 the -k (or --insecure) option.
HTTPS-proxy has similar options --proxy-cacert and --proxy-insecure.
```
depending if hitting the https or http endpoint.

```
kubectl version
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.0", GitCommit:"2bd9643cee5b3b3a5ecbd3af49d09018f0773c77", GitTreeState:"clean", BuildDate:"2019-09-19T13:57:45Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.2", GitCommit:"c97fe5036ef3df2967d086711e6c0c405941e14b", GitTreeState:"clean", BuildDate:"2019-10-15T19:09:08Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}
```

Created using Rancher --> Add Cluster --> Linode. 
Did this with a very basic setup and one with split etcd, CP, and 3 workers.

**Comment by StrongMonkey** [comment]
I think this is a problem that we have a conflict with existing ingress controller that rke creates.

**Comment by daxmc99** [comment]
Verified that rio installs correctly into a RKE created v1.16 when using the cluster endpoint and bypassing the rancher-proxy entirely. I was able to run the rio demo successfully. 

This fails in k3d/ k3s v0.10.0 with unable to pull the localhost image. 
```
rio ps
NAME             IMAGE                          ENDPOINT                                                  SCALE     APP              VERSION    WEIGHT    CREATED              DETAIL
bold-williams2   default-bold-williams2:f8fab   http://bold-williams2-fc0df4f0-default.n2r6kn.on-rio.io   0/1 0%    bold-williams2   fc0df4f0   100%      About a minute ago   bold-williams2: bold-williams2(ImagePullBackOff/Back-off pulling image "localhost:5442/default-bold-williams2:f8fab")
```

**Comment by ibuildthecloud** [comment]
This looks like an upstream issue with containerd CRI.  We'll get a new k3s release for this. https://github.com/containerd/cri/pull/1328

**Comment by StrongMonkey** [comment]
Available to test in v0.6.0-alpha2. @rancher-max When you test this, make sure all RKE cluster that created by rancher should work as expected.

**Comment by rancher-max** [comment]
default backend 404 has been resolved in v0.6.0-alpha.2 on k8s linode cluster created through Rancher. Leaving issue open to track rio side for original ImagePullBackOff issue

**Comment by rancher-max** [comment]
Fixed with k3s v0.10.2: https://github.com/rancher/k3s/releases/tag/v0.10.2
